src/OrleansAzureUtils/Hosting/AzureSilo.cs

```diff
@@ -61,13 +71,43 @@ public AzureSilo()
             logger = LogManager.GetLogger("OrleansAzureSilo", LoggerType.Runtime);
         }
 
+        public async Task<bool> ValidateConfiguration(ClusterConfiguration config)
+        {
+            if (config.Globals.LivenessType == GlobalConfiguration.LivenessProviderType.AzureTable)
+            {
+                string deploymentId = serviceRuntimeWrapper.DeploymentId;
+                string connectionString = serviceRuntimeWrapper.GetConfigurationSettingValue(DataConnectionConfigurationSettingName);
+
+                try
+                {
+                    var manager = await OrleansSiloInstanceManager.GetManager(deploymentId, connectionString);
+                    var instances = await manager.DumpSiloInstanceTable();
+                }
+                catch (Exception exc)
+                {
+                    var error = String.Format("The connection string is incorrect. Connecting to the storage table has failed with {0}",
```

Comment:
We don't know "The connection string is incorrect."  We just know the operation failed.